### PR TITLE
[Chat][Feature] Chat demo app

### PR DESCRIPTION
**Pull request by Mrayyan**
## Purpose

Demo app for chat. Launch button launches chat activity.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/chat/build.gradle [hunk 1]
```diff
@@ -28,16 +31,38 @@ android {
...
+    }
+    packagingOptions {
+        resources {
+            excludes += '/META-INF/{AL2.0,LGPL2.1}'
+        }
+    }
 }
 
 dependencies {
+    api project(path: ':calling')
```
Comment (pavelprystinka) [inline azure-communication-ui/chat/build.gradle:48]: Chat should not depend on calling

File: azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/CachedTokenFetcher.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (pavelprystinka) [inline azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/CachedTokenFetcher.kt:1]: You can place classes like Fetcher in the shared location in the main folder.
See UrlTokenFetcher for example

File: azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/CallLauncherActivityErrorHandler.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (pavelprystinka) [inline azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/CallLauncherActivityErrorHandler.kt:1]: We do not need Call in the Chat

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatComposite.java [hunk 4]
```diff
@@ -3,5 +3,16 @@
...
 public class ChatComposite {
+
+    ChatComposite() {
+    }
+
+    public void launch(final Context context) {
+        final Intent launchIntent = new Intent(context, ChatActivity.class);
+        context.startActivity(launchIntent);
+    }
 }
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatComposite.java:18]: @Mrayyan this should be separate PR

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ChatCompositeActivity.kt [hunk 5]
```diff
@@ -4,15 +4,34 @@
...
+            ) {
+                BasicText(
+                    text = "Hello UI Library Chat!",
+                    modifier = Modifier
+                        .align(Alignment.CenterHorizontally)
+                )
+            }
         }
     }
 }
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ChatCompositeActivity.kt:37]: We should not modify chat Ui library activity as part of this PR

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/controlbar/ControlBarView.kt [hunk 6]
```diff
@@ -101,8 +101,8 @@ internal class ControlBarView : ConstraintLayout {
 
     private val alwaysOffSelectedAccessibilityDelegate = object : AccessibilityDelegateCompat() {
         override fun onInitializeAccessibilityNodeInfo(
-            host: View?,
-            info: AccessibilityNodeInfoCompat?
+            host: View,
+            info: AccessibilityNodeInfoCompat
```
Comment (iaulakh) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/controlbar/ControlBarView.kt:105]: would it be possible to fix this as separate PR if required as this is not part of chat

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+package com.azure.android.communication.ui.chat;
+
+import com.azure.android.communication.ui.chat.configuration.ChatCompositeConfiguration;
+import com.azure.android.communication.ui.chat.models.ChatCompositeLocalizationOptions;
+
+/**
+ * Builder for creating {@link ChatComposite}.
+ *
+ * <p>Used to build a {@link ChatComposite} which is then used to start a call.</p>
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java:12]: call

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+ * <p>This class can be used to specify a locale to be used by the Chat Composite</p>
+ */
+public final class ChatCompositeBuilder {
+
+    private ChatCompositeLocalizationOptions localizationConfig = null;
+
+    /**
+     * Sets an optional localization for call-composite to use by {@link ChatComposite}.
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java:20]: chat

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+     * @param localization {@link ChatCompositeLocalizationOptions}.
+     * @return {@link ChatCompositeBuilder} for chaining options
+     */
+    public ChatCompositeBuilder localization(final ChatCompositeLocalizationOptions localization) {
+        this.localizationConfig = localization;
+        return this;
+    }
+
+    /**
+     * Builds the CallCompositeClass {@link ChatComposite}.
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/ChatCompositeBuilder.java:31]: Chat

File: azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/ChatLauncherActivity.kt [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+            if (savedInstanceState.getBoolean(isKotlinLauncherOptionSelected)) {
+                callLauncherViewModel.setKotlinLauncher()
+            } else {
+                callLauncherViewModel.setJavaLauncher()
+            }
+        }
+
+        binding.run {
+            launchButton.setOnClickListener {
+                callLauncherViewModel.doLaunch()
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/ChatLauncherActivity.kt:66]: chatLauncherViewModel

File: azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/SettingsActivity.kt [hunk 11]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.chatdemoapp
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.azure.android.communication.ui.callingcompositedemoapp.R
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/SettingsActivity.kt:8]: chatcompositedemoapp

File: azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/features/AdditionalFeatures.kt [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.chatdemoapp.features
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import com.azure.android.communication.ui.callingcompositedemoapp.R
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/java/com/azure/android/communication/ui/chatdemoapp/features/AdditionalFeatures.kt:11]: chatdemoapp

File: azure-communication-ui/demo-app/src/chat/res/layout/activity_chat_launcher.xml [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    >
+
+    <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:paddingStart="6dp"
+        android:paddingEnd="6dp"
+        tools:context=".CallLauncherActivity"
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/res/layout/activity_chat_launcher.xml:17]: ChatLauncherActivity

**Review by iaulakh — CHANGES_REQUESTED** [review]

File: azure-communication-ui/demo-app/src/chat/AndroidManifest.xml [hunk 14]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+   Copyright (c) Microsoft Corporation. All rights reserved.
+   Licensed under the MIT License.
+  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.azure.android.communication.ui.callingcompositedemoapp"
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/AndroidManifest.xml:9]: demoapp

File: azure-communication-ui/demo-app/src/chat/AndroidManifest.xml [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:host="https://test"
+                    android:pathPrefix="/calling"
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/AndroidManifest.xml:47]: chat

File: azure-communication-ui/demo-app/src/chat/AndroidManifest.xml [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+                    android:scheme="https"
+                    />
+                <data
+                    android:host="https://test"
+                    android:pathPrefix="/calling"
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/chat/AndroidManifest.xml:52]: chat

**Review by iaulakh — APPROVED** [review]

**Review by Mohtasim — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]
